tests/lava/proc/plateau/test_models.py

```diff
@@ -0,0 +1,178 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+
+import unittest
+import numpy as np
+from numpy.testing import assert_almost_equal
+
+from lava.magma.core.decorator import implements, requires, tag
+from lava.magma.core.model.py.model import PyLoihiProcessModel
+from lava.magma.core.model.py.ports import PyOutPort, PyInPort
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.process.ports.ports import OutPort, InPort
+from lava.magma.core.process.process import AbstractProcess
+from lava.magma.core.process.variable import Var
+from lava.magma.core.resources import CPU
+from lava.proc.plateau.process import Plateau
+from lava.proc.dense.process import Dense
+from lava.magma.core.run_configs import Loihi2SimCfg, RunConfig
+from lava.magma.core.run_conditions import RunSteps
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.tests.lava.proc.lif.test_models import VecSendProcess, VecRecvProcess
+
+
+class SpikeGen(AbstractProcess):
+    """Process for sending spikes at user-supplied time steps.
+
+    Parameters
+    ----------
+    spikes_in: list[list], list of lists containing spike times
+    runtime: int, number of timesteps for the generator to store spikes
+    """
+    def __init__(self, spikes_in, runtime):
+        super().__init__()
+        n = len(spikes_in)
+        self.shape = (n,)
+        spike_data = np.zeros(shape=(n, runtime))
+        for i in range(n):
+            for t in range(1, runtime + 1):
+                if t in spikes_in[i]:
+                    spike_data[i, t - 1] = 1
+        self.s_out = OutPort(shape=self.shape)
+        self.spike_data = Var(shape=(n, runtime), init=spike_data)
+
+
+@implements(proc=SpikeGen, protocol=LoihiProtocol)
+@requires(CPU)
+@tag('fixed_pt')
+class PySpikeGenModel(PyLoihiProcessModel):
+    s_out: PyOutPort = LavaPyType(PyOutPort.VEC_DENSE, float)
+    spike_data: np.ndarray = LavaPyType(np.ndarray, float)
+
+    def run_spk(self):
+        """Send the appropriate spikes for the given time step
+        """
+        self.s_out.send(self.spike_data[:, self.time_step - 1])
```

Comment:
One of the advantages of event-based, neuromorphic computing is that we don't need to communicate '0's. Thus, could you check whether the SpikeGen actually sends a meaningful information at the respective time step, and ensure not to send anything instead of a 0?